src/mds/Server.cc

```diff
@@ -7749,6 +7758,21 @@ void Server::handle_client_rename(MDRequestRef& mdr)
     return;
   }
 
+  if (req->get_data().length()) {
+    auto p = req->get_data().cbegin();
+
+    // xattrs on new inode?
+    auto _xattrs = CInode::allocate_xattr_map();
+    decode_noshare(*_xattrs, p);
+
+    // The "fscrypt.alternate_name" is not a real
+    // xattr, will remove it from the xattrs map.
+    auto it = _xattrs->find("fscrypt.alternate_name");
+    if (it != _xattrs->end())
+      destdn->set_alternate_name(std::move(it->second));
+    dout(10) << "the new name length is a long name!" << dendl;
+  }
+
```

Comment:
I see this part for renames, but what about hardlink creation? Those don't create a new inode, but they do create a dentry. Do we need similar handling there?

Reply:
Yeah, we need, I will fix it.